src/http/client.cr

```diff
@@ -380,94 +380,96 @@ class HTTP::Client
+  {% for http_method in %w(post put patch) %}
+    # Executes a {{http_method.id.upcase}} with form data. The "Content-type" header is set
+    # to "application/x-www-form-urlencoded".
+    #
+    # ```
+    # client = HTTP::Client.new "www.example.com"
+    # response = client.{{http_method.id}}_form "/", "foo=bar"
+    # ```
+    def {{http_method.id}}_form(path, form : String | IO, headers : HTTP::Headers? = nil) : HTTP::Client::Response
+      request = new_request({{http_method.upcase}}, path, headers, form)
+      request.headers["Content-type"] = "application/x-www-form-urlencoded"
+      exec request
+    end
 
-  # Executes a POST with form data and yields the response to the block.
-  # The response will have its body as an `IO` accessed via `HTTP::Client::Response#body_io`.
-  # The "Content-type" header is set to "application/x-www-form-urlencoded".
-  #
-  # ```
-  # client = HTTP::Client.new "www.example.com"
-  # client.post_form("/", "foo=bar") do |response|
-  #   response.body_io.gets
-  # end
-  # ```
-  def post_form(path, form : String | IO, headers : HTTP::Headers? = nil)
-    request = new_request("POST", path, headers, form)
-    request.headers["Content-type"] = "application/x-www-form-urlencoded"
-    exec(request) do |response|
-      yield response
+    # Executes a {{http_method.id.upcase}} with form data and yields the response to the block.
+    # The response will have its body as an `IO` accessed via `HTTP::Client::Response#body_io`.
+    # The "Content-type" header is set to "application/x-www-form-urlencoded".
+    #
+    # ```
+    # client = HTTP::Client.new "www.example.com"
+    # client.{{http_method.id}}_form("/", "foo=bar") do |response|
+    #   response.body_io.gets
+    # end
+    # ```
+    def {{http_method.id}}_form(path, form : String | IO, headers : HTTP::Headers? = nil)
+      request = new_request({{http_method.upcase}}, path, headers, form)
+      request.headers["Content-type"] = "application/x-www-form-urlencoded"
+      exec(request) do |response|
+        yield response
+      end
     end
-  end
 
-  # Executes a POST with form data. The "Content-type" header is set
-  # to "application/x-www-form-urlencoded".
-  #
-  # ```
-  # client = HTTP::Client.new "www.example.com"
-  # response = client.post_form "/", {"foo" => "bar"}
-  # ```
-  def post_form(path, form : Hash(String, _) | NamedTuple, headers : HTTP::Headers? = nil) : HTTP::Client::Response
-    body = HTTP::Params.encode(form)
-    post_form path, body, headers
-  end
+    # Executes a {{http_method.id.upcase}} with form data. The "Content-type" header is set
+    # to "application/x-www-form-urlencoded".
+    #
+    # ```
+    # client = HTTP::Client.new "www.example.com"
+    # response = client.{{http_method.id}}_form "/", {"foo" => "bar"}
+    # ```
+    def {{http_method.id}}_form(path, form : Hash(String, _) | NamedTuple, headers : HTTP::Headers? = nil) : HTTP::Client::Response
+      body = HTTP::Params.from_hash(form)
```

Comment:
C) `HTTP::Params.from_hash` has been renamed to `HTTP::Params.encode` (#4205). That's why the specs are failing.